airflow/bin/cli.py

```diff
@@ -541,18 +930,138 @@ def upgradedb(args):  # noqa
     print("DB: " + repr(settings.engine.url))
     db_utils.upgradedb()
 
+    # Populate DagStats table
+    session = settings.Session()
+    ds_rows = session.query(DagStat).count()
+    if not ds_rows:
+        qry = (
+            session.query(DagRun.dag_id, DagRun.state, func.count('*'))
+            .group_by(DagRun.dag_id, DagRun.state)
+        )
+        for dag_id, state, count in qry:
+            session.add(DagStat(dag_id=dag_id, state=state, count=count))
+        session.commit()
+
 
 def version(args):  # noqa
     print(settings.HEADER + "  v" + airflow.__version__)
 
 
+def connections(args):
+    if args.list:
+        # Check that no other flags were passed to the command
+        invalid_args = list()
+        for arg in ['conn_id', 'conn_uri', 'conn_extra']:
+            if getattr(args, arg) is not None:
+                invalid_args.append(arg)
+        if invalid_args:
+            msg = ('\n\tThe following args are not compatible with the ' +
+                   '--list flag: {invalid!r}\n')
+            msg = msg.format(invalid=invalid_args)
+            print(msg)
+            return
+
+        session = settings.Session()
+        conns = session.query(Connection.conn_id, Connection.conn_type,
+                              Connection.host, Connection.port,
+                              Connection.is_encrypted,
+                              Connection.is_extra_encrypted,
+                              Connection.extra).all()
+        conns = [map(reprlib.repr, conn) for conn in conns]
+        print(tabulate(conns, ['Conn Id', 'Conn Type', 'Host', 'Port',
+                               'Is Encrypted', 'Is Extra Encrypted', 'Extra'],
+                       tablefmt="fancy_grid"))
+        return
+
+    if args.delete:
```

Comment:
I don't think we really use connections here, but we can add them via the cli now we want to